lib/appsignal/integrations/capistrano/appsignal.cap

```diff
@@ -5,19 +5,27 @@ namespace :appsignal do
   task :deploy do
     appsignal_env = fetch(:appsignal_env,
       fetch(:stage, fetch(:rails_env, fetch(:rack_env, "production"))))
-    user = fetch(:appsignal_user, ENV["USER"] || ENV.fetch("USERNAME", nil))
-    revision = fetch(:appsignal_revision, fetch(:current_revision))
 
     Appsignal._load_config!(appsignal_env) do |config|
       config&.merge_dsl_options(fetch(:appsignal_config, {}))
     end
     Appsignal._start_logger
 
     if Appsignal.config&.active?
-      marker_data = {
-        :revision => revision,
-        :user => user
-      }
+      # Check if REVISION file exists - if so, skip setting revision and user
+      revision_file_path = File.join(fetch(:deploy_to, Dir.pwd), "REVISION")
+      if File.exist?(revision_file_path) && File.readable?(revision_file_path)
+        marker_data = {}
+        puts "REVISION file found at #{revision_file_path}, " \
+          "skipping Capistrano revision/user reporting"
+      else
+        user = fetch(:appsignal_user, ENV["USER"] || ENV.fetch("USERNAME", nil))
+        revision = fetch(:appsignal_revision, fetch(:current_revision))
+        marker_data = {
+          :revision => revision,
+          :user => user
+        }
+      end
```

Comment:
The problem with this is, I think, that the REVISION file is only available on the deployed app's host. It is not available locally, which is where this script runs.

Unless we remove the two config options completely, we can't ensure only `REVISION` is used. Which is a breaking change I don't want to do right now.

Let's revert this change, and the similar one for Capistrano 2, and let's see how it will work.

If revision is read from the REVISION file, each sample is tagged with the revision, that's an improvement that automagically would work.
If configured, we'd still report the revision this way too. That's fine for a first release of this change.